keras/callbacks.py

```diff
@@ -2733,6 +2774,13 @@ def on_train_batch_end(self, batch, logs=None):
                 1.0 / batch_run_time,
                 step=self._train_step,
             )
+
+        # `logs` is a `RemoteValue` when using asynchronous strategies, for now
+        # we just disable `update_freq` entirely in those cases.
+        if isinstance(logs, dict):
```

Comment:
Sorry, there's also a performance implication for TPUStrategy as that's asynchronous and reading the metrics at batch interval forces a sync. Can you also exclude that here (by raising an error if using TPUStrategy)?

Reply:
I'd argue that the performance degradation is already implied by the concept of `update_freq` and can be avoided by setting a larger value so that syncs don't happen as often, although I'm not 100% sure how tensorflow works here: does `tf.summary.scalar` only read the values for logging if `tf.summary.record_if` is true?

Reply:
Yes, that's my understanding.